components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -434,231 +409,172 @@ void AdsServiceImpl::OnMigrateConfirmationState(const bool success) {
     return;
```

Comment:
If we failed to migrate a confirmation state, then a device id won't be retreived. Maybe we need to call GetDeviceId() even if migration failed?

Reply:
If we fail confirmation, state migration ads won't run, so the device id is unnecessary. Did I miss something?

Reply:
Fixed. Thanks